### DISCUSSION TITLE
Pydantic validation error 'loc' does not generate valid openapi spec.

### DISCUSSION THREAD

**Discussion by kevgliss** [fastapi/fastapi#8767] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List
from fastapi import FastAPI
from pydantic import BaseModel
from pprint import pprint

from fastapi.openapi.utils import get_openapi

app = FastAPI()


class Item(BaseModel):
    name: str


@app.post("/items/")
async def read_items(item: List[Item]):
    return {"item": item}


openapi_schema = get_openapi(
    title="Custom title",
    version="2.5.0",
    description="This is a very custom OpenAPI schema",
    routes=app.routes,
)
pprint(openapi_schema)
```

### Description

The pydantic error location results are defined as `Tuple[Union[int, str], ...]` here:
https://github.com/samuelcolvin/pydantic/blob/4a54f393ad20ee91b51cd7a49ec46771ba4f8a18/pydantic/error_wrappers.py#L12

However, the open api schema generated only renderers as an array of type string. This means that the spec does not validate against the returned object when pydantic returns an integer. 

As an example, the following tests would not pass spec validation. 

https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/tests/test_multi_body_errors.py#L74

https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/tests/test_multi_body_errors.py#L115

### Operating System

Linux

### FastAPI Version

0.67.0

### Python Version

Python 3.8.7

**Comment by anthonycepeda** [comment]
Hi @kevgliss 

Let's say you change your Item.name field for an integer in your model:
```python
class Item(BaseModel):
    name: int
```
And then you try pass an string to your /items endpoint:```[{ "name": "test" }] ```
You will have the following `422` error:
```javascript
{
  "detail": [
    {
      "loc": [
        "body",
        0,
        "name"
      ],
      "msg": "value is not a valid integer",
      "type": "type_error.integer"
    }
  ]
}
```

This means that even when (I guess for simplicity), the  `Tuple[Union[int, str], ...]` type is not been shown in the Openapi definitions, FastAPI raises the correct error.

**NOTE** If your type for `Item.name` is string, when you introduce an integer, it will be converted to string by your Pydatic Model automatically, hence no error.

Thanks

**Comment by papasax** [comment]
Hi,

I wen't into the same problem testing my API with schemathesis. It won't pass the tests because of this.
```python
class Item(BaseModel):
    id: UUID
```
FastAPI generates this error:

```json
{
   "loc":[
      "body",
      0,
      "id"
   ],
   "msg":"value is not a valid uuid",
   "type":"type_error.uuid"
}
```

and schemathesis this one:

```
1. The received response does not conform to the defined schema!
Details: 
0 is not of type 'string'
Failed validating 'type' in schema['properties']['detail']['items']['properties']['loc']['items']:
    {'type': 'string'}
On instance['detail'][1]['loc'][1]:
```

I ended up patching the generated OpenAPI Schema within schemathesis:
```python
schema = schemathesis.from_asgi("/openapi.json", application_provider())

schema.raw_schema['components']['schemas']['ValidationError']['properties']['loc']['items'] = {"oneOf":[
    {"type":"string"},
    {"type":"number"},
]}
```

I think this could be done within FastAPI.

Regards,

**Comment by YuriiMotov** [comment]
That was fixed
